File: scripts/checkAnalysis.js
import fs from 'fs';
import path from 'path';

const lessonsDir = './src/data/lessons';
const files = fs.readdirSync(lessonsDir)
    .filter(f => f.startsWith('nce1-l') && f.endsWith('.json'))
    .sort((a, b) => {
        const numA = parseInt(a.match(/-l(\d+)/)[1]);
        const numB = parseInt(b.match(/-l(\d+)/)[1]);
        return numA - numB;
    });

const results = [];

files.forEach(file => {
    const filePath = path.join(lessonsDir, file);
    const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));

    let totalWordsInText = 0;
    let totalAnalyzedWords = 0;

    data.segments.forEach(segment => {
        if (segment.text && !segment.id.startsWith('intro')) {
            const wordsInText = segment.text.replace(/['",.!?]/g, '').trim().split(/\s+/).filter(w => w.length > 0).length;
            totalWordsInText += wordsInText;

            if (segment.analysis && segment.analysis.words) {
                totalAnalyzedWords += segment.analysis.words.length;
            }
        }
    });

    const wordRatio = totalWordsInText > 0 ? totalAnalyzedWords / totalWordsInText : 0;
    const lessonNum = parseInt(file.match(/-l(\d+)/)[1]);

    if (wordRatio < 0.9 && lessonNum <= 131) {
        results.push({
            file,
            lessonNum,
            wordRatio: wordRatio.toFixed(2)
        });
    }
});

console.log(JSON.stringify(results, null, 2));

File: check_analysis.py

import json
import os
import re

def count_words(text):
    return len(re.findall(r'\w+', text))

lessons_dir = 'src/data/lessons'
incomplete_lessons = []

for filename in sorted(os.listdir(lessons_dir)):
    if filename.startswith('nce2-l') and filename.endswith('.json'):
        filepath = os.path.join(lessons_dir, filename)
        with open(filepath, 'r') as f:
            data = json.load(f)
            total_missing = 0
            total_sparse = 0
            for segment in data.get('segments', []):
                text_word_count = count_words(segment.get('text', ''))
                analysis_word_count = len(segment.get('analysis', {}).get('words', []))
                
                if analysis_word_count == 0:
                    total_missing += 1
                elif analysis_word_count < text_word_count * 0.5:
                    total_sparse += 1
            
            if total_missing > 0 or total_sparse > 0:
                incomplete_lessons.append({
                    'file': filename,
                    'missing': total_missing,
                    'sparse': total_sparse
                })

for item in incomplete_lessons:
    print(f"{item['file']}: {item['missing']} missing, {item['sparse']} sparse")
